common/src/api/external/mod.rs

```diff
@@ -1207,162 +1264,66 @@ pub struct VpcRouter {
     pub vpc_id: Uuid,
 }
 
-/// Represents all possible network target strings as defined in RFD-21
-/// This enum itself isn't intended to be used directly but rather as a
-/// delegate for subset enums to not have to re-implement all the base type conversions.
-///
-/// See <https://rfd.shared.oxide.computer/rfd/0021#api-target-strings>
-#[derive(Debug, PartialEq, Display, FromStr)]
-pub enum NetworkTarget {
-    #[display("vpc:{0}")]
-    Vpc(Name),
-    #[display("subnet:{0}")]
-    Subnet(Name),
-    #[display("instance:{0}")]
-    Instance(Name),
-    #[display("tag:{0}")]
-    Tag(Name),
-    #[display("ip:{0}")]
-    Ip(IpAddr),
-    #[display("inetgw:{0}")]
-    InternetGateway(Name),
-    #[display("fip:{0}")]
-    FloatingIp(Name),
-}
-
-/// A subset of [`NetworkTarget`], `RouteTarget` specifies all
-/// possible targets that a route can forward to.
-#[derive(Clone, Debug, Deserialize, Serialize, PartialEq, JsonSchema)]
+/// A `RouteTarget` describes the possible locations that traffic matching a
+/// route destination can be sent.
+#[derive(
+    Clone,
+    Debug,
+    Deserialize,
+    Display,
+    FromStr,
+    Serialize,
+    PartialEq,
+    JsonSchema,
+)]
 #[serde(tag = "type", content = "value", rename_all = "snake_case")]
+#[display(style = "lowercase")]
 pub enum RouteTarget {
+    /// Forward traffic to a particular IP address.
+    #[display("{}:{0}")]
     Ip(IpAddr),
+    /// Forward traffic a VPC
+    #[display("{}:{0}")]
     Vpc(Name),
+    /// Forward traffic a VPC Subnet
+    #[display("{}:{0}")]
     Subnet(Name),
+    /// Forward traffic a specific instance
+    #[display("{}:{0}")]
     Instance(Name),
+    #[display("inetgw:{0}")]
```

Comment:
If you defined the top level display and left this one, would this take president or would it just fail?

Reply:
Good question. I wasn't sure, so did the safe thing. But I'll try it out now :)

Reply:
Cool! It looks like the enum-level attribute is used for most variants, but can be overridden per-variant. Very nice, will push.